File: easy/1518_water_bottles/solution.py
class Solution:
    def numWaterBottles(self, num_bottles: int, num_exchanges: int) -> int:
        total_num_bottles = num_empty_bottles = 0

        while num_bottles:
            total_num_bottles += num_bottles
            num_empty_bottles += num_bottles

            num_bottles = num_empty_bottles // num_exchanges
            num_empty_bottles %= num_exchanges

        return total_num_bottles

File: easy/1518_water_bottles/solution.cpp
class Solution {
public:
  int numWaterBottles(int numBottles, int numExchanges) {
    int totalNumBottles = 0;
    int numEmptyBottles = 0;

    while (numBottles) {
      totalNumBottles += numBottles;
      numEmptyBottles += numBottles;

      numBottles = numEmptyBottles / numExchanges;
      numEmptyBottles %= numExchanges;
    }

    return totalNumBottles;
  }
};
